Modules/_testcapi/watchers.c

```diff
@@ -724,6 +724,15 @@ clear_context_stack(PyObject *Py_UNUSED(self), PyObject *Py_UNUSED(args))
     Py_RETURN_NONE;
 }
 
+static PyObject *
+context_enter(PyObject *self, PyObject *ctx)
+{
+    if (PyContext_Enter(ctx)) {
```

Comment:
`PyContext_Enter` and `PyContext_Exit` use 0 and -1 for the return values. In C, both of those are falsy, so failures won't correctly propagate.
```suggestion
    if (PyContext_Enter(ctx) < 0) {
```

Reply:
> In C, both of those are falsy

Anything non-zero is truthy (C23 6.8.4.1p2).  To reduce the likelihood of introducing a security vulnerability, best practice is to look for non-success (`!= 0`) rather than look for error (`== -1`, or `< 0`).

Reply:
We don't support C23, do we? I think we compile for C11

Reply:
No, I was just using that as a reference.  The standard hasn't changed in this regard since the first version was published.

Following convention is a good enough reason, so I'll change them.